hibernate-reactive-core/src/test/java/org/hibernate/reactive/ReactiveMultitenantNoResolverTest.java

```diff
@@ -266,74 +270,80 @@ public void testOpenStatelessSessionWithTenantWithMutiny(TestContext context) {
-		thrown.expectCause( isA( HibernateException.class ) );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getSessionFactory().withStatelessTransaction( (s, t) -> selectCurrentDB( s ) ) );
+		test( context, ReactiveAssertions.assertThrown( HibernateException.class, getSessionFactory().withStatelessTransaction( this::selectCurrentDB ) )
+				.thenAccept( exception -> Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" ) ) );
 	}
 
 	@Test
 	public void testOpenSessionThrowsExceptionWithoutTenantWithMutiny(TestContext context) {
-		thrown.expect( HibernateException.class );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, openMutinySession().call( this::selectCurrentDB ) );
+		test( context, ReactiveAssertions.assertThrown(
+				HibernateException.class, openMutinySession().invoke( this::selectCurrentDB ) )
+				.onFailure().recoverWithUni( exception -> {
+					Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" );
+					return Uni.createFrom().nullItem();
+				} )
+		);
 	}
 
 	@Test
 	public void testWithSessionThrowsExceptionWithoutTenantWithMutiny(TestContext context) {
-		thrown.expect( HibernateException.class );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getMutinySessionFactory().withSession( this::selectCurrentDB ) );
+		test( context, ReactiveAssertions.assertThrown(
+				HibernateException.class, getMutinySessionFactory().withSession( this::selectCurrentDB ) )
+				.onFailure().recoverWithUni( exception -> {
+					Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" );
+					return Uni.createFrom().nullItem();
+				} )
+		);
 	}
 
 	@Test
 	public void testWithTransactionThrowsExceptionWithoutTenantWithMutiny(TestContext context) {
-		thrown.expect( HibernateException.class );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getMutinySessionFactory().withTransaction( (s, t) -> selectCurrentDB( s ) ) );
+		test( context, ReactiveAssertions.assertThrown(
+				HibernateException.class, getMutinySessionFactory().withTransaction( this::selectCurrentDB ) )
+				.onFailure().recoverWithUni( exception -> {
+					Assertions.assertThat( exception.getMessage() ).contains( "no tenant identifier" );
+					return Uni.createFrom().nullItem();
+				} )
+		);
 	}
 
 	@Test
 	public void testWithStatelessSessionThrowsExceptionWithoutTenant(TestContext context) {
-		thrown.expectCause( isA( HibernateException.class ) );
-		thrown.expectMessage( "no tenant identifier" );
-
-		test( context, getSessionFactory().withStatelessSession( this::selectCurrentDB ) );
+		test( context, ReactiveAssertions.assertThrown( HibernateException.class, getSessionFactory().withStatelessTransaction( this::selectCurrentDB ) )
+				.thenAccept( e -> ReactiveAssertions.assertException( e, HibernateException.class,
```

Comment:
Non need to assert the exception twice, you can just assert the message

Reply:
refactored and fixed